cmd/cloudshell_open/cloudrun.go

```diff
@@ -87,12 +87,14 @@ func promptDeploymentRegion(ctx context.Context, project string) (string, error)
 	return choice, nil
 }
 
-func serviceURL(project, name, region string) (string, error) {
+
+func describe(project, name, region, format, flatten string) (string, error) {
 	cmd := exec.Command("gcloud", "run", "services", "describe", name,
 		"--project", project,
 		"--platform", "managed",
 		"--region", region,
-		"--format", "value(status.address.url)")
+		"--format", format,
```

Comment:
IMO let's just do `--format=json` here and parse it into a legit Service object and return it. This is opening a large can of worms (i.e. Test_describe) and complicated `--format` strings.

Reply:
Yeah, that makes sense.  I think I'll just have to create a bunch of structs to hold the fairly large tree.  Or can I use Knative as a library?

Reply:
I think we can very simply create a tiny new struct that has fields for

- spec.containers[].env[]
- status.address.url (or whatever, I forgot)

Importing Knative struct here isn't gonna do much good tbh, since we don't use most of it. But it'll increase the binary size significantly since it imports all Kubernetes API machinery code.

Maybe the cloudrun api clients we have actually have this type.

In that case, maybe we can even eliminate call to `gcloud run services describe` with a proper GetService API call.

Reply:
Yeah, it would be nice to use the client APIs.  I can't find a way for `gcloud` `format=json` to give me a child, like:
```
gcloud run services describe hello-netcat --platform=managed --project=jw-demo --region=us-central1 --format="json(spec.template)"
{
  "spec": {
    "template": {
      "metadata": {
        "annotations": {
          "client.knative.dev/user-image": "gcr.io/jw-demo/hello-netcat",
          "run.googleapis.com/client-name": "gcloud",
          "run.googleapis.com/client-version": "272.0.0"
        },
        "name": "hello-netcat-00003-vaq"
      },
      "spec": {
        "containerConcurrency": 0,
        "containers": [
          {
            "image": "gcr.io/jw-demo/hello-netcat",
            "resources": {
              "limits": {
                "memory": "512Mi"
              }
            }
          }
        ]
      }
    }
  }
}
```

Reply:
e.g. `format="json(spec.template)"` reduces the tree but doesn't crop it.